src/unicode/utf8/utf8_spec.gobra

```diff
@@ -0,0 +1,70 @@
+package utf8
+//+gobra
+
+import (
+	. "verification/utils/definitions"
+	sl "gobra-libs/byteslice"
+)
+
+// utf8.Codepoints(s) returns the sequence of (utf-8 encoded) codepoints in `s`.
+// Handles invalid utf-8 by replacing offending bytes with RuneError.
+//
+// See utf8.DecodeRune
+ghost
+pure
+ensures 0 < len(s) ==> 0 < len(res)
+ensures 0 == len(s) ==> 0 == len(res)
+trusted
+decreases
+func Codepoints(s []byte) (res seq[rune])
+
+
+const (
+	RuneError = '\uFFFD'
+	RuneSelf  = 0x80
+	MaxRune   = '\U0010FFFF'
+	UTFMax    = 4
+)
+
+// AppendRune appends the UTF-8 encoding of r to the end of p and
+// returns the extended buffer. If the rune is out of range,
+// it appends the encoding of [RuneError].
+func AppendRune(p []byte, r rune) []byte
+
+preserves acc(sl.Bytes(p, 0, len(p)), R40)
+ensures len(p) > 0 ==> 1 <= size && size <= len(p)
+ensures len(p) == 0 ==> size == 0
+ensures len(p) > 0 ==> Codepoints(p)[0] == r && Codepoints(p[size:]) == Codepoints(p)[1:]
```

Comment:
To make the spec more readable, please add a line break right after `==>` on long implications, and introduce a tab before the rhs on the next line.